### PR TITLE
Token Balances - Fix incorrect CoinGecko IDs for built in network assets

### LINKED ISSUE #3116: BNB balance shown incorrectly
### Discord Discussion Link

https://discord.com/channels/808358975287722045/1037670242421321769/1081898920868859904

### What browsers are you seeing the problem on?

Chrome

### What were you trying to do?

0x4fb1b0C4728c51c9c9f8791B30400990e0520316
This member's address has 0.00795633 BNB on the chain, but it does not show up when added to the wallet via read-only mode, probably a bug.

### What did not work?

N/A

### Version

v0.27.0

### Relevant log output

_No response_

### LINKED ISSUE #3122: Cannot sign typed data requests
### Discord Discussion Link

_No response_

### What browsers are you seeing the problem on?

Chrome

### What were you trying to do?

image

### What did not work?

Both buttons are disabled

### Version

v0.27.1

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by hyphenized**
Corrects CoinGecko token IDs for network base assets, also fixes an issue with the signing buttons for typed data requests.

Closes #3116
Closes #3122

## To Test
- [ ] Open the extension, switch to bsc, balance in USD for BNB should display correctly

Latest build: [extension-builds-3120](https://github.com/tahowallet/extension/suites/11452762012/artifacts/590796956) (as of Thu, 09 Mar 2023 12:13:23 GMT).

### CODE REVIEW

File: ui/components/Shared/__tests__/SharedAssetIcon.test.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import React from "react"
+import { render } from "@testing-library/react"
+import SharedAssetIcon from "../SharedAssetIcon"
+
+describe("SharedAssetIcon", () => {
+  test("smoke test", async () => {
```
Comment (kkosiorowska) [inline ui/components/Shared/__tests__/SharedAssetIcon.test.tsx:6]:
```suggestion
  test("should render asset icon ", () => {
```

File: ui/components/Shared/__tests__/SharedAssetIcon.test.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        symbol="USDC"
+        logoURL="http://localhost:8097/supercoolassetlogo.jpg"
+      />
+    )
+
+    expect(ui.getByRole("img")).toBeInTheDocument()
+    expect(ui.getByRole("img")).toBeVisible()
+  })
+
+  test("should handle assets with invalid symbols", async () => {
```
Comment (kkosiorowska) [inline ui/components/Shared/__tests__/SharedAssetIcon.test.tsx:19]:
```suggestion
  test("should handle assets with invalid symbols", () => {
```

File: ui/components/Shared/__tests__/SharedAssetIcon.test.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    const ui = render(
+      <SharedAssetIcon
+        size="small"
+        symbol=""
+        logoURL="http://localhost:8097/supercoolassetlogo.jpg"
+      />
+    )
+
+    expect(ui.getByRole("img")).toBeInTheDocument()
+    expect(ui.getByRole("img")).toBeVisible()
```
Comment (kkosiorowska) [inline ui/components/Shared/__tests__/SharedAssetIcon.test.tsx:29]:
```suggestion
    expect(ui.getByRole("img")).toBeInTheDocument()
    expect(ui.getByRole("img")).toBeVisible()
    expect(ui.getByText("?")).toBeVisible()
```

**Review by kkosiorowska — COMMENTED** [review]
I haven't tested it yet. Some minor comments.

**Comment by jagodarybacka** [comment]
Fix for signing buttons has been cherry-picked and tested, the rest of this PR should go through the regular path so @hyphenized will have time to address comments

File: ui/components/Signing/Signer/TransactionButton.tsx [hunk 4]
```diff
@@ -17,9 +20,11 @@ export default function TransactionButton({
 }: SharedButtonProps & { reactOnWindowFocus?: boolean }): ReactElement {
   const hasTransactionLoaded = useBackgroundSelector(selectIsTransactionLoaded)
 
-  const hasSignDataRequest = useBackgroundSelector(selectSigningData)
+  const signDataRequest = useBackgroundSelector(selectSigningData)
+  const typedDataRequest = useBackgroundSelector(selectTypedData)
+  const hasSigningRequest = signDataRequest || typedDataRequest
 
-  const isTransactionDataReady = hasTransactionLoaded || hasSignDataRequest
+  const isTransactionDataReady = hasTransactionLoaded || hasSigningRequest
```
[the remark this answers is not in the document]
Comment (hyphenized) [reply ui/components/Signing/Signer/TransactionButton.tsx:27]: Would prefer to address this in a separate PR

**Pull request merged**